src/redux/dataLoad.js

```diff
@@ -0,0 +1,17 @@
+import {updateLoadState, updateData} from './peopleSlice.js';
+
+export async function dataLoad(dispatch) {
+  try {
+    dispatch(updateLoadState({state: true, error: null}));
+    const response = await fetch('https://swapi.py4e.com/api/people');
+    if (response.ok) {
+      const data = await response.json();
+      dispatch(updateLoadState({state: true, error: null}));
+      dispatch(updateData(data));
+    } else {
+      dispatch(updateLoadState({state: false, error: 'HTTP error ' + response.status}));
+    }
+  } catch (err) {
+    dispatch(updateLoadState({state: false, error: err.message}));
+  }
+}
```

Comment:
1. Move this function to `helpers.js`.
2. Change the interface to `dataLoad(dispatch, updateLoadState, updateData, url)`
3. Import new function to this file
4. Reexport new function with required args:
```js
import { updateLoadState, updateData } from './peopleSlice.js';
import { dataLoad } from './helpers.js';
 
export async function loadPeopleData(dispatch) {
  return dataLoad(dispatch, updateLoadState, updateData, 'https://swapi.py4e.com/api/people');
}
```